modules/features2d/test/test_sift.cpp

```diff
@@ -0,0 +1,107 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+
+namespace opencv_test { namespace {
+
+class CV_SiftTest : public cvtest::BaseTest
+{
+public:
+    CV_SiftTest();
+    ~CV_SiftTest();
+protected:
+    void run(int);
+};
+
+CV_SiftTest::CV_SiftTest() {}
+CV_SiftTest::~CV_SiftTest() {}
+
+void CV_SiftTest::run( int )
+{
+    Mat image = imread(string(ts->get_data_path()) + "features2d/tsukuba.png");
+    string xml = string(ts->get_data_path()) + "sift/result.xml.gz";
+
+    if (image.empty())
+    {
+        ts->printf( cvtest::TS::LOG, "No image.\n" );
+        ts->set_failed_test_info( cvtest::TS::FAIL_INVALID_TEST_DATA );
+        return;
+    }
+
+    Mat gray;
+    cvtColor(image, gray, COLOR_BGR2GRAY);
+
+    vector<KeyPoint> calcKeypoints;
+    Mat calcDescriptors;
+    Ptr<SIFT> sift = cv::SIFT::create();
+    sift->detectAndCompute(gray, Mat(), calcKeypoints, calcDescriptors, false);
+
+    FileStorage fs(xml, FileStorage::READ);
+    if (!fs.isOpened())
+    {
```

Comment:
Implicit data re-generation is dangerous.

```
#ifndef GENERATE_DATA
    ASSERT_TRUE(fs.isOpened()) << xml;
    ... read ...
#else
    ... creating XML ...
#endif // GENERATE_DATA
```

Commented `//#define GENERATE_DATA` in the beginning of the file (see test_houghlines.cpp as example)